gdnative-core/src/core_types/variant.rs

```diff
@@ -1881,6 +1883,294 @@ macro_rules! impl_variant_for_tuples {
 
 impl_variant_for_tuples!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12,);
 
+#[cfg(feature = "serde")]
+pub mod serde {
+    use super::super::{
+        dictionary::serde::DictionaryVisitor, variant_array::serde::VariantArrayVisitor,
+    };
+    use super::*;
+    use ::serde::{
+        de::{EnumAccess, Error, MapAccess, SeqAccess, Visitor},
+        Deserialize, Deserializer, Serialize, Serializer,
+    };
+    use std::fmt::Formatter;
+
+    impl Serialize for Variant {
+        #[inline]
+        fn serialize<S>(
+            &self,
+            serializer: S,
+        ) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+        where
+            S: Serializer,
+        {
+            serializer.serialize_newtype_struct("Variant", &self.dispatch())
+        }
+    }
+
+    struct VariantDispatchVisitor;
+
+    impl<'de> Visitor<'de> for VariantDispatchVisitor {
```

Comment:
This whole deserialize impl is questionable since it doesn't match what serde derives for the struct at all. This visitor is attempting to deserialize an untagged enum, but `#[derive(Serialize)]` without any further attributes actually produces an externally-tagged one. See https://serde.rs/enum-representations.html for more info.

Reply:
Ah, I see. As I said, still learning serde and that's why I knew I should open the PR to get feedback early.